### ISSUE TITLE
lint: consider to use rwmutex against mutex, in readonly sections

### ISSUE THREAD

**Issue by fexolm** [go-critic/go-critic#276]

**Comment by NexoMichael** [comment]
* The RWMutex-ex are not really needed from my experience with new processors/oses.
* Current implementation has scale issues https://github.com/golang/go/issues/17973

**Comment by cristaloleg** [comment]
> RWMutex-ex are not really needed

But how do you synchronise your flow? 🤔  (if it's needed of course)

**Comment by Quasilyte** [comment]
> But how do you synchronise your flow? 🤔 (if it's needed of course)

With "normal" `sync.Mutex`, I guess.

**Comment by NexoMichael** [comment]
Yes, simple 'sync.Mutex' to determine real pros for 'sync.RWMutex' to use need some normal testing with accurate calculation of read/write ratio.
